src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/MuTect2.java

```diff
@@ -0,0 +1,1057 @@
+            return referenceModelForNoVariation(originalAssemblyRegion, true);
+        } else if( originalAssemblyRegion.size() == 0 ) {
+            return referenceModelForNoVariation(originalAssemblyRegion, true);
+        }
+        logReadInfo(DEBUG_READ_NAME, originalAssemblyRegion.getReads(), "Present in original active region");
+
+        // create the assembly using just high quality reads (Q20 or higher).  We want to use lower
+        // quality reads in the PairHMM (and especially in the normal) later, so we can't use a ReadFilter
+        final AssemblyRegion assemblyAssemblyRegion = new AssemblyRegion(originalAssemblyRegion.getSpan(), originalAssemblyRegion.getSupportingStates(), originalAssemblyRegion.isActive(), originalAssemblyRegion.getExtension(), header);
+        for (final GATKRead rec : originalAssemblyRegion.getReads()) {
+            if (rec.getMappingQuality() >= MQthreshold ) {
+                assemblyAssemblyRegion.add(rec);
+            }
+        }
+        logReadInfo(DEBUG_READ_NAME, assemblyAssemblyRegion.getReads(), "Present in assembly active region");
+
+        // run the local assembler, getting back a collection of information on how we should proceed
+        final AssemblyResultSet untrimmedAssemblyResult = assembleReads(assemblyAssemblyRegion, Collections.emptyList());
+        final SortedSet<VariantContext> allVariationEvents = untrimmedAssemblyResult.getVariationEvents();
+        final AssemblyRegionTrimmer.Result trimmingResult = trimmer.trim(originalAssemblyRegion,allVariationEvents);
+        if (!trimmingResult.isVariationPresent()) {
+            return referenceModelForNoVariation(originalAssemblyRegion, false);
+        }
+        logReadInfo(DEBUG_READ_NAME, trimmingResult.getCallableRegion().getReads(), "Present in trimming result");
+
+        final AssemblyResultSet assemblyResult =
+                trimmingResult.needsTrimming() ? untrimmedAssemblyResult.trimTo(trimmingResult.getCallableRegion()) : untrimmedAssemblyResult;
+
+        // it is conceivable that the region is active yet has no events upon assembling only the well-mapped reads
+        if( ! assemblyResult.isVariationPresent() ) {
+            return referenceModelForNoVariation(originalAssemblyRegion, false);
+        }
+
+        final AssemblyRegion regionForGenotyping = assemblyResult.getRegionForGenotyping();
+        logReadInfo(DEBUG_READ_NAME, regionForGenotyping.getReads(), "Present in region for genotyping");
+
+        // filter out reads from genotyping which fail mapping quality based criteria
+        //TODO - why don't do this before any assembly is done? Why not just once at the beginning of this method
+        //TODO - on the originalAssemblyRegion?
+        //TODO - if you move this up you might have to consider to change referenceModelForNoVariation
+        //TODO - that does also filter reads.
+        final Collection<GATKRead> filteredReads = filterNonPassingReads(regionForGenotyping);
```

Comment:
I could believe that explanation.  If that's the case we may be able to improve things by using somatic-specific parameters for assembly instead of borrowing them from `HaplotypeCaller`.